// ng-zorro-antd/esm2020/water-mark/water-mark.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzWaterMarkComponent } from './water-mark.component';
import * as i0 from "@angular/core";
export class NzWaterMarkModule {
}
NzWaterMarkModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzWaterMarkModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NzWaterMarkModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.5", ngImport: i0, type: NzWaterMarkModule, declarations: [NzWaterMarkComponent], imports: [CommonModule], exports: [NzWaterMarkComponent] });
NzWaterMarkModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzWaterMarkModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzWaterMarkModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [NzWaterMarkComponent],
                    declarations: [NzWaterMarkComponent],
                    imports: [CommonModule]
                }]
        }] });
