// ng-zorro-antd/esm2020/tabs/tabs.module.mjs
/**
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
 */
import { A11yModule } from '@angular/cdk/a11y';
import { BidiModule } from '@angular/cdk/bidi';
import { ObserversModule } from '@angular/cdk/observers';
import { PlatformModule } from '@angular/cdk/platform';
import { CdkScrollableModule } from '@angular/cdk/scrolling';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzIconModule } from 'ng-zorro-antd/icon';
import { NzTabAddButtonComponent } from './tab-add-button.component';
import { NzTabBodyComponent } from './tab-body.component';
import { NzTabCloseButtonComponent } from './tab-close-button.component';
import { NzTabLinkDirective, NzTabLinkTemplateDirective } from './tab-link.directive';
import { NzTabNavBarComponent } from './tab-nav-bar.component';
import { NzTabNavItemDirective } from './tab-nav-item.directive';
import { NzTabNavOperationComponent } from './tab-nav-operation.component';
import { NzTabScrollListDirective } from './tab-scroll-list.directive';
import { NzTabComponent } from './tab.component';
import { NzTabDirective } from './tab.directive';
import { NzTabsInkBarDirective } from './tabs-ink-bar.directive';
import { NzTabSetComponent } from './tabset.component';
import * as i0 from "@angular/core";
const DIRECTIVES = [
    NzTabSetComponent,
    NzTabComponent,
    NzTabNavBarComponent,
    NzTabNavItemDirective,
    NzTabsInkBarDirective,
    NzTabScrollListDirective,
    NzTabNavOperationComponent,
    NzTabAddButtonComponent,
    NzTabCloseButtonComponent,
    NzTabDirective,
    NzTabBodyComponent,
    NzTabLinkDirective,
    NzTabLinkTemplateDirective
];
export class NzTabsModule {
}
NzTabsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NzTabsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.5", ngImport: i0, type: NzTabsModule, declarations: [NzTabSetComponent,
        NzTabComponent,
        NzTabNavBarComponent,
        NzTabNavItemDirective,
        NzTabsInkBarDirective,
        NzTabScrollListDirective,
        NzTabNavOperationComponent,
        NzTabAddButtonComponent,
        NzTabCloseButtonComponent,
        NzTabDirective,
        NzTabBodyComponent,
        NzTabLinkDirective,
        NzTabLinkTemplateDirective], imports: [BidiModule,
        CommonModule,
        ObserversModule,
        NzIconModule,
        NzOutletModule,
        PlatformModule,
        A11yModule,
        CdkScrollableModule,
        NzDropDownModule], exports: [NzTabSetComponent,
        NzTabComponent,
        NzTabNavBarComponent,
        NzTabNavItemDirective,
        NzTabsInkBarDirective,
        NzTabScrollListDirective,
        NzTabNavOperationComponent,
        NzTabAddButtonComponent,
        NzTabCloseButtonComponent,
        NzTabDirective,
        NzTabBodyComponent,
        NzTabLinkDirective,
        NzTabLinkTemplateDirective] });
NzTabsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzTabsModule, imports: [BidiModule,
        CommonModule,
        ObserversModule,
        NzIconModule,
        NzOutletModule,
        PlatformModule,
        A11yModule,
        CdkScrollableModule,
        NzDropDownModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.5", ngImport: i0, type: NzTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DIRECTIVES],
                    exports: [DIRECTIVES],
                    imports: [
                        BidiModule,
                        CommonModule,
                        ObserversModule,
                        NzIconModule,
                        NzOutletModule,
                        PlatformModule,
                        A11yModule,
                        CdkScrollableModule,
                        NzDropDownModule
                    ]
                }]
        }] });
